clients/da-vinci-client/src/main/java/com/linkedin/davinci/replication/merge/MapKeyStringAnnotatedStoreSchemaCache.java

```diff
@@ -0,0 +1,98 @@
+package com.linkedin.davinci.replication.merge;
+
+import static com.linkedin.venice.schema.SchemaUtils.getAnnotatedStringMapDerivedSchemaEntry;
+import static com.linkedin.venice.schema.SchemaUtils.getAnnotatedStringMapValueSchemaEntry;
+
+import com.linkedin.venice.meta.ReadOnlySchemaRepository;
+import com.linkedin.venice.schema.SchemaEntry;
+import com.linkedin.venice.schema.writecompute.DerivedSchemaEntry;
+import com.linkedin.venice.utils.concurrent.VeniceConcurrentHashMap;
+import java.util.Map;
+import java.util.Optional;
+
+
+/**
+ * This class is a wrapper schema repository class, which is only used by merge conflict resolver in A/A store ingestion
+ * task. This class will annotate value schema and partial update schema so when performing Map collection merging,
+ * Map field's key can be deserialized into String type and thus is comparable for DCR purpose. Without this wrapper or
+ * user annotation, map fields will have key in UTF-8 type which is not comparable.
+ * This wrapper class only implements retrieval of the value schema (including the latest value schema and superset schema)
+ * and partial update schema as they are the only usage in merge conflict resolver. Other operations are not supported
+ * intentionally to avoid unexpected behavior.
+ */
+public class MapKeyStringAnnotatedStoreSchemaCache {
+  private final ReadOnlySchemaRepository internalSchemaRepo;
+  private final String storeName;
+  private final Map<Integer, SchemaEntry> valueSchemaEntryMapCache = new VeniceConcurrentHashMap<>();
+  private final Map<String, DerivedSchemaEntry> partialUpdateSchemaEntryMapCache = new VeniceConcurrentHashMap<>();
+
+  public MapKeyStringAnnotatedStoreSchemaCache(String storeName, ReadOnlySchemaRepository internalSchemaRepo) {
+    this.storeName = storeName;
+    this.internalSchemaRepo = internalSchemaRepo;
+  }
+
+  public void clear() {
+    valueSchemaEntryMapCache.clear();
+    partialUpdateSchemaEntryMapCache.clear();
+    internalSchemaRepo.clear();
```

Comment:
This is not right.
The passed `ReadOnlySchemaRepository` is the SoT and each `MapKeyStringAnnotatedStoreSchemaCache` is just a partial view of it and when clearing an instance of `MapKeyStringAnnotatedStoreSchemaCache`, we shouldn't touch the shared schema repo.

Reply:
Given the fact that this method is never used anywhere...I think we should just remove this method.